pac/source-repo/docker-pull-request.yaml

```diff
@@ -16,32 +17,32 @@ metadata:
 spec:
   params:
     - name: dockerfile
-      value: {{ values.dockerfile }}
+      value: {{values.dockerfile}}
```

Comment:
I copied this PR branch and pushed it to my remote [here](https://github.com/Jdubrick/rhdh-pipelines/tree/yaml-lint), and then updated the reference in one of my branches of the ai-lab-template [here](https://github.com/Jdubrick/ai-lab-template/tree/yaml-lint-testing) to pull the changes in. I noticed that it alters the formatting for anything the linter didn't surround in `'` to be a ternary? I'm confident this will break the pipelines but will check it anyway

cc @jrichter1 @maysunfaisal 

![Screenshot 2024-10-28 at 1 47 53 PM](https://github.com/user-attachments/assets/1d5dfde5-7bd8-4e8e-868a-3fce105ccac5)

Reply:
From first glance however I notice that the linter in `ai-lab-app` isn't messing with the spacing inside of our `{{ values.<whatever> }}` inserts

Reply:
> From first glance however I notice that the linter in `ai-lab-app` isn't messing with the spacing inside of our `{{ values.<whatever> }}` inserts. Most likely from [redhat-ai-dev/ai-lab-app@main...jrichter1:ai-lab-app:yaml-lint#diff-944291df2c9c06359d37cc8833d182d705c9e8c3108e7cfe132d61a06e9133ddR18](https://github.com/redhat-ai-dev/ai-lab-app/compare/main...jrichter1:ai-lab-app:yaml-lint#diff-944291df2c9c06359d37cc8833d182d705c9e8c3108e7cfe132d61a06e9133ddR18)

I don't think there was a single instance of spacing inside curly braces in `ai-lab-app`

Reply:
> I copied this PR branch and pushed it to my remote [here](https://github.com/Jdubrick/rhdh-pipelines/tree/yaml-lint), and then updated the reference in one of my branches of the ai-lab-template [here](https://github.com/Jdubrick/ai-lab-template/tree/yaml-lint-testing) to pull the changes in. I noticed that it alters the formatting for anything the linter didn't surround in `'` to be a ternary? I'm confident this will break the pipelines but will check it anyway

Turns out the generate script expects spacing within the braces. The `sed` command can't match double braces without the space, so it wouldn't add the `$` character. Then `yq` just goes berserk and changes the values into this weird contraption.

The question now is, spaces, or no spaces?

Reply:
You're right, there is no spacing in the curly braces in `ai-lab-app`, I was mistakenly looking at the `ai-lab-app` portion of `ai-lab-template` where it formats the content in `ai-lab-app` to have spaces in the curlies. I was looking at the Backstage docs and they have the syntax of `{{ <value> }}` with the spaces so I would say we go with spaces then - It's fine for the linter to not include the spaces for `ai-lab-app` I think since our `./generate.sh` script formats them to have spaces

Reply:
reverted to a singular space inside, though `{{values.rawUrl}}` remains unchanged because the generate script doesn't expect a space there